packages/replay/src/types.ts

```diff
@@ -18,6 +18,17 @@ export interface SendReplayData {
   options: ReplayPluginOptions;
 }
 
+export enum FlushState {
```

Comment:
We should generally avoid enums as they have a pretty hefty bundle size impact ([more details](https://blog.sentry.io/2022/07/22/performance-impact-of-generated-javascript/#switching-from-typescript-enums-to-const-and-string-enums))

We could use string constants instead, something along these lines:
```typescript
export type FlushState = 'start' | 'sent-request' | 'complete' | 'error'
```
WDYT?